invite/test/invite_bot.test.ts

```diff
@@ -66,32 +67,33 @@ describe('Invite Bot', () => {
 
   describe('constructor', () => {
     it('defaults helpUserTag to "someone"', () => {
-      inviteBot = new InviteBot(/*client=*/ null, 'test_org');
+      inviteBot = new InviteBot(/*client=*/ null, 'test_org', 'wg-example');
       expect(inviteBot.helpUserTag).toEqual('someone in your organization');
     });
 
     it('prepends the help username with @ if set', () => {
       inviteBot = new InviteBot(
         /*client=*/ null,
         'test_org',
-        /*helpUsernameToTag=*/'test_org/wg-example',
+        'wg-example',
+        /*helpUsernameToTag=*/'test_org/wg-helpme',
       );
-      expect(inviteBot.helpUserTag).toEqual('@test_org/wg-example');
+      expect(inviteBot.helpUserTag).toEqual('@test_org/wg-helpme');
     });
   });
 
   describe('processComment', () => {
     beforeEach(() => {
+      jest.spyOn(inviteBot, 'userCanTrigger');
```

Comment:
Should've noticed this in the previous round of reviews... As a rule of thumb don't spy on or mock things that are inside the class/module that you're testing - spy or mock _only_ its dependencies. You can still `expect(method()).toBe / .resolves.toBe`, but instead of `expect(method()).toBeCalledWith / .not.toBeCalled` you'll have to replace this with its dependencies.

When you spy on your own methods you create a tight coupling between the code under test, and the test itself. When you mock your own methods you are testing that your test is testing itself correctly. When you mock the dependencies you create a tight coupling between the test and the code's dependencies, which is better because ideally you use stable dependencies which themselves shouldn't change :)

(there is, of course, no perfect way to decouple the test from the code entirely, but the less friction between the test and the code that is being tested, the better)

Reply:
This was written before I'd implemented the underlying functionality; I'll swap it out for a spy on the GitHub instance.

Reply:
Done.